### ISSUE TITLE
Global variable $10 makes ruby-lint crash, without reporting the offending node location

### ISSUE THREAD

**Issue by mvidner** [YorickPeterse/ruby-lint#175]
```console
$ cat dollar-number.rb 
s = "1234567890"
r = /(.)(.)(.)(.)(.)(.)(.)(.)(.)(.)/
s =~ r
puts $10
$ ruby -w dollar-number.rb
0
$ ruby-lint dollar-number.rb
/local/.../ruby-lint-2.2.0/lib/ruby-lint/virtual_machine.rb:356:in `on_nth_ref': undefined method `value' for nil:NilClass (NoMethodError)
        from /local/.../ruby-lint-2.2.0/lib/ruby-lint/iterator.rb:125:in `execute_callback'
        from /local/.../ruby-lint-2.2.0/lib/ruby-lint/iterator.rb:86:in `block in iterate'
        from /local/.../ruby-lint-2.2.0/lib/ruby-lint/iterator.rb:85:in `catch'
...
```

I was quite surprised to discover that this is valid Ruby :)

Here the crash is a bit more problematic than in #173 because the location of the problematic source is not reported.

Originally found on https://github.com/crowbar/crowbar-openstack/blob/5644aff965526e77db669ec8373dc4e026140035/chef/cookbooks/swift/providers/ringfile.rb#L132
